### DISCUSSION TITLE
Bottom half of panel is green

### DISCUSSION THREAD

**Discussion by michaelnaman** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#532] [Q&A]
I have several panels with 64x32 RUC7258D RUL2038S 16s but only the top half the panels are working, the bottom half is green.

![PXL_20231201_015341861](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/185379/7b9a882b-dbfa-4ae9-b416-5614bb414ea3)

The chips on the back read RUC7258D RUL2038S  which I believe is supported. 

However the HUB75 pins is not the standard layout, note the B2 and G2 switched position.
// R1 | G1
// B1 | GND
// R2 | B2
// G2 | GND
//  A | B
//  C | D
// CLK| LAT
// OE | GND

I have also had to swap around the RGB pins R1 = G1, G1 = B1, B1 = R1 pins to get solid colours to display correctly on the top part of the screen that is working. Swapping the R2, etc pins around makes no difference and still displays green.

Is there anything I can do to get the lower half of the screen working? This is going to be a christmas present for my son and am really up against it time wise so any help you can give would be greatly appreciated.

**Comment by foorschtbar** [comment]
I got the same panels, but mine keeps complete black. Have a working matrix and replaced only one panel, this keep black. Setup a single ESP which works fine on my "old" panels, but the new one with RUC7258D & RUL2038S keep black.

**Comment by ergindemir** [comment]
Have you tried protomatter library, if it works then it's an issue with this library. I have a 6126A module that is completely blank and works fine with protomatter.

**Comment by foorschtbar** [comment]
With mxconfig.driver = HUB75_I2S_CFG::ICN2038S; now it works :)

**Comment by michaelnaman** [comment]
Thanks for the answers! It's been put away in it's box for Christmas but I'll get it back out and try again tonight and post the results.

**Comment by michaelnaman** [comment]
@foorschtbar  Thanks for the answer, I've downloaded the latest from github and installed it. 

I have the full panel working but only every other line.

![PXL_20240102_211947645 MP](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/185379/a00b5ee9-c707-45ae-95bf-ee2c5ffadeaf)

This is the current set up code

`  HUB75_I2S_CFG::i2s_pins _pins={R1_PIN, G1_PIN, B1_PIN, R2_PIN, G2_PIN, B2_PIN, A_PIN, B_PIN, C_PIN, D_PIN, E_PIN, LAT_PIN, OE_PIN, CLK_PIN};
  HUB75_I2S_CFG mxconfig(64, 32, 1, _pins);
  mxconfig.driver = HUB75_I2S_CFG::ICN2038S; 
  dma_display = new MatrixPanel_I2S_DMA(mxconfig);`

**Comment by foorschtbar** [comment]
I copy some lines i use, maybe they help you:

Lib Version i use: mrfaptastic/ESP32 HUB75 LED MATRIX PANEL DMA Display@^2.0.7

```c
[...]
#include <ESP32-HUB75-MatrixPanel-I2S-DMA.h>
#include <ESP32-VirtualMatrixPanel-I2S-DMA.h>
[...]

#define PANEL_RES_X 64 // Number of pixels wide of each INDIVIDUAL panel module.
#define PANEL_RES_Y 32 // Number of pixels tall of each INDIVIDUAL panel module.

#define PANEL_CHAIN NUM_ROWS *NUM_COLS // total number of panels chained one to another

// Change this to your needs, for details on VirtualPanel pls read the PDF!
#define SERPENT true
#define TOPDOWN true

#define R1_PIN 27 // 25
#define G1_PIN 25 // 26
#define B1_PIN 26 // 27
#define R2_PIN 13 // 14
#define G2_PIN 14 // 12
#define B2_PIN 12 // 13
#define A_PIN 23
#define B_PIN 19
#define C_PIN 5
#define D_PIN 17
#define E_PIN -1 // required for 1/32 scan panels, like 64x64px. Any available pin would do, i.e. IO32
#define LAT_PIN 4
#define OE_PIN 15
#define CLK_PIN 16

HUB75_I2S_CFG::i2s_pins _pins = {R1_PIN, G1_PIN, B1_PIN, R2_PIN, G2_PIN, B2_PIN, A_PIN, B_PIN, C_PIN, D_PIN, E_PIN, LAT_PIN, OE_PIN, CLK_PIN};

// placeholder for the matrix object
MatrixPanel_I2S_DMA *dma_display = nullptr;

// placeholder for the virtual display object
VirtualMatrixPanel *virtualDisp = nullptr;

void startDMADisplay()
{
  HUB75_I2S_CFG mxconfig(
      PANEL_RES_X, // module width
      PANEL_RES_Y, // module height
      PANEL_CHAIN, // chain length
      _pins        // ESP32 pin configuration
  );

  mxconfig.driver = HUB75_I2S_CFG::ICN2038S; // use the ICN2038S driver chip

  // OK, now we can create our matrix object
  dma_display = new MatrixPanel_I2S_DMA(mxconfig);

  // let's adjust default brightness to about 75%
  dma_display->setBrightness8(currentBrightness); // range is 0-255, 0 - 0%, 255 - 100%

  // Allocate memory and start DMA display
  logln(F("Start DMA Display..."));
  if (not dma_display->begin())
    logln(F("I2S memory allocation failed"));

  // create VirtualDisplay object based on our newly created dma_display object
  virtualDisp = new VirtualMatrixPanel((*dma_display), NUM_ROWS, NUM_COLS, PANEL_RES_X, PANEL_RES_Y, SERPENT, TOPDOWN);

  displayIsReady = true;

  logln(F("First screen test..."));
  screenNumbering();
  // screenTest();
}

void setup()
{

 [...]

  // Start screen stuff
  startDMADisplay();

 [...]

  logln(F("Setup done."));

}

```
